### PR TITLE
feat: add a badge to the notifications menu Item

### PR DESCRIPTION
**Pull request by ducklin5**
#### Summary of changes/features added:
- Notifications Fragment uses the profile available by RootActivity
- Notification menu item shows a badge with the count if the there are any notifications
- Menu Item updates on page change 
    - this need to be improved so that menu items update when the profile is updated
    - maybe through a profile.onUpdate listener 

#### Screenshots of UI changes (update with each new commit): 
CriEngineRootActivity4

## Pull Request Checklist
- [ ] Unit test written?
- [ ] (**For later in project**) Integration tests written?
- [x] Java docs written?

### CODE REVIEW

File: client/app/src/main/java/com/example/criengine/Activities/RootActivity.java [hunk 1]
```diff
@@ -51,21 +66,25 @@ protected void onCreate(Bundle savedInstanceState) {
         // If returning from another activity, this can control which screen to navigate to.
         if (getIntent().getExtras() != null) {
             int index = ((PAGE) getIntent().getSerializableExtra("Index")).getValue();
-            viewPager.setCurrentItem(index);
+            viewPager.setCurrentItem(index, false);
         } else {
             // Returns to my books. This will be the home screen.
-            viewPager.setCurrentItem(PAGE.MY_BOOKS.getValue());
+            viewPager.setCurrentItem(PAGE.MY_BOOKS.getValue(), false);
```
Comment (ducklin5) [inline client/app/src/main/java/com/example/criengine/Activities/RootActivity.java:72]: `false` changes the current page without animation/scrolling.
It looks weird when this Activity opens up on page 0 and immediately scrolls to `index` or `MY_BOOKS`

Comment (AdityaHarvi) [reply client/app/src/main/java/com/example/criengine/Activities/RootActivity.java:72]: Ayy thats what we needed. ty

File: client/app/src/main/java/com/example/criengine/Activities/RootActivity.java [hunk 2]
```diff
@@ -97,10 +116,30 @@ public void onPageSelected(int position) {
...
+     */
+    private void updateNotificationBadge() {
+        dbw.getProfile(dbw.userId).addOnSuccessListener(
+                new OnSuccessListener<Profile>() {
+                    @Override
+                    public void onSuccess(Profile profile) {
+                        int notificationCount = dummyProfile.getNotifications().size();
+                        BadgeDrawable badge = navigation
+                                .getOrCreateBadge(R.id.bottom_navigation_item_notifications);
+                        // TODO: uncomment the next line when notifications work
```
Comment (AdityaHarvi) [inline client/app/src/main/java/com/example/criengine/Activities/RootActivity.java:135]: What line must be uncommented? Nothing after this is commented lmao

Comment (ducklin5) [reply client/app/src/main/java/com/example/criengine/Activities/RootActivity.java:135]: Oh potatoes, I forgot to remove this

Comment (ducklin5) [reply client/app/src/main/java/com/example/criengine/Activities/RootActivity.java]: fixed

File: client/app/src/main/java/com/example/criengine/Activities/RootActivity.java [hunk 3]
```diff
@@ -51,21 +66,25 @@ protected void onCreate(Bundle savedInstanceState) {
...
     private class RootPagerFragmentAdapter extends FragmentStateAdapter {
-        public RootPagerFragmentAdapter(@NonNull FragmentActivity fa) { super(fa); }
+        public RootPagerFragmentAdapter(@NonNull FragmentActivity fa) {
+            super(fa);
+        }
 
         @Override
-        public int getItemCount() { return 5; }
+        public int getItemCount() {
+            return 5;
```
Comment (shadowciaw) [inline client/app/src/main/java/com/example/criengine/Activities/RootActivity.java:86]: smells kind of magic number. any way to change this to something else?

**Review by AdityaHarvi — APPROVED** [review]

File: client/app/src/main/java/com/example/criengine/Activities/RootActivity.java (hunk 3 above)
Comment (ducklin5) [reply client/app/src/main/java/com/example/criengine/Activities/RootActivity.java:118]: alrighty, I'll add it as a constant to the class

**Review by shadowciaw — APPROVED** [review]

File: client/app/src/main/java/com/example/criengine/Activities/RootActivity.java (hunk 3 above)
Comment (ducklin5) [reply client/app/src/main/java/com/example/criengine/Activities/RootActivity.java]: Is that alright?

Comment (shadowciaw) [reply client/app/src/main/java/com/example/criengine/Activities/RootActivity.java]: no way to get something like PAGE.length? it's still a magic number, you just moved it out

**Pull request merged**